terraform/modules/cds_export/access_log_bucket.tf

```diff
@@ -0,0 +1,101 @@
+#tfsec:ignore:aws-s3-enable-bucket-encryption: default bucket encryption is sufficient
+#tfsec:ignore:aws-s3-enable-bucket-logging: access log bucket doesn't need access logging itself
+#tfsec:ignore:aws-s3-encryption-customer-key: default encryption is sufficient
+resource "aws_s3_bucket" "export_access_logs" {
+  #checkov:skip=CKV2_AWS_62: no need for event notifications
+  #checkov:skip=CKV_AWS_18: access log bucket doesn't need access logging itself
+  #checkov:skip=CKV_AWS_145: default encryption is fine
+  #checkov:skip=CKV_AWS_144: cross region replication not required for access logs
+  bucket = "${var.prefix}-export${var.bucket_suffix}-logs"
+}
+
+resource "aws_s3_bucket_public_access_block" "export_access_logs" {
+  bucket = aws_s3_bucket.export_access_logs.id
+
+  block_public_acls       = true
+  block_public_policy     = true
+  ignore_public_acls      = true
+  restrict_public_buckets = true
+}
+
+data "aws_caller_identity" "current" {}
+
+resource "aws_s3_bucket_policy" "force_ssl_and_allow_access_logs" {
+  bucket = aws_s3_bucket.export_access_logs.id
+
+  policy = jsonencode({
+    Version = "2012-10-17",
+    Statement = [
+      {
+        Sid       = "AllowSSLRequestsOnly",
+        Action    = "s3:*",
+        Effect    = "Deny",
+        Principal = "*",
+        Resource = [
+          aws_s3_bucket.export_access_logs.arn,
+          "${aws_s3_bucket.export_access_logs.arn}/*"
+        ],
+        Condition = {
+          Bool = {
+            "aws:SecureTransport" = "false"
+          }
+        }
+      },
+      {
+        Sid    = "AllowS3AccessLogs",
+        Action = "s3:PutObject",
+        Effect = "Allow",
+        Principal = {
+          "Service" = "logging.s3.amazonaws.com"
+        },
+        Resource = [
+          "${aws_s3_bucket.export_access_logs.arn}/*"
+        ],
+        Condition = {
+          ArnLike = {
+            "aws:SourceArn" = aws_s3_bucket.export.arn
+          },
+          StringEquals = {
+            "aws:SourceAccount" = data.aws_caller_identity.current.account_id
+          }
+        }
+      }
+    ]
+  })
+}
+
+resource "aws_s3_bucket_versioning" "export_access_logs" {
+  bucket = aws_s3_bucket.export_access_logs.id
+
+  versioning_configuration {
+    status = "Enabled"
+  }
+}
+
+resource "aws_s3_bucket_lifecycle_configuration" "export_access_logs" {
+  bucket = aws_s3_bucket.export_access_logs.id
+
+  rule {
+    id     = "expire-old-logs"
+    status = "Enabled"
+
+    abort_incomplete_multipart_upload {
+      days_after_initiation = 1
+    }
+
+    expiration {
+      days = 90
+    }
+
+    filter {}
+
+    noncurrent_version_expiration {
+      noncurrent_days = 90
+    }
+
+    noncurrent_version_transition {
+      noncurrent_days = 30
+      storage_class   = "STANDARD_IA"

```

Comment:
Since we're doing bucket versioning, it seemed prudent to move non-current versions of log files into a cheaper storage class in case they hang around. I picked `STANDARD_IA` as it seemed the best balance of cost / availability, but you could probably argue a case for all the storage classes listed [here](https://aws.amazon.com/s3/storage-classes/)!